src/ray/core_worker/transport/direct_task_transport.cc

```diff
@@ -15,14 +15,39 @@
 #include "ray/core_worker/transport/direct_task_transport.h"
 
 #include "ray/core_worker/transport/dependency_resolver.h"
-#include "ray/core_worker/transport/direct_actor_transport.h"
 
 namespace ray {
 
 Status CoreWorkerDirectTaskSubmitter::SubmitTask(TaskSpecification task_spec) {
   RAY_LOG(DEBUG) << "Submit task " << task_spec.TaskId();
   resolver_.ResolveDependencies(task_spec, [this, task_spec]() {
     RAY_LOG(DEBUG) << "Task dependencies resolved " << task_spec.TaskId();
+    if (RayConfig::instance().gcs_service_enabled() && task_spec.IsActorCreationTask()) {
+      RAY_CHECK(gcs_client_ != nullptr);
+      // If gcs actor management is enabled, the actor creation task will be sent to gcs
+      // server directly after the in-memory dependent objects are resolved.
+      // For more details please see the protocol of actor management based on gcs.
+      // https://docs.google.com/document/d/1EAWide-jy05akJp6OMtDn58XOK7bUyruWMia4E-fV28/edit?usp=sharing
+      auto actor_id = task_spec.ActorCreationId();
+      RAY_LOG(INFO) << "Submitting actor creation task to GCS: " << actor_id;
+      auto status =
+          gcs_client_->Actors().AsyncCreateActor(task_spec, [actor_id](Status status) {
+            if (status.ok()) {
+              RAY_LOG(INFO) << "Actor creation task submitted to GCS: " << actor_id;
+            } else {
+              // If GCS is failed, GcsRpcClient may receive IOError status but it will not
+              // trigger this callback, because GcsRpcClient has retry logic at the
+              // bottom. So it must be something wrong with the protocol of gcs-based
```

Comment:
> Why not put the assertion check into GcsRpcClient instead?

That's because the callback of AsyncCreateActor is a lambda with `Status`, so I need to handle all of the case.